### ISSUE TITLE
Custom loss function

### ISSUE THREAD

**Issue by davidlkl** [ZebinYang/gaminet#3]
Hi,

I find this paper very useful and looking to apply it for my use case.

Do you have plans to extend GAMI-Net to include tweedie loss etc.? Or to allow custom loss function and metrics just like the other packages (e.g. xgboost).

Thanks!

**Comment by ZebinYang** [comment]
Hi, 

Thanks for your interest in our paper. 
Currently, we have no plan to extend GAMI-Net with other kinds of losses. 
If you would like to use a custom loss, a possible solution is to overwrite "self.loss_fn" after initializing GAMINet. For example,
the default MSE can be changed to MeanAbsoluteError,

model = GAMINet(meta_info=meta_info, interact_num=20, interact_arch=[20, 10], subnet_arch=[20, 10],
                task_type=task_type, activation_func=tf.tanh, main_grid_size=41, interact_grid_size=41,
                batch_size=500, lr_bp=0.001, main_effect_epochs=2000,
                interaction_epochs=2000, tuning_epochs=50, loss_threshold=0.01,
                verbose=True, val_ratio=0.2, early_stop_thres=100)
model.loss_fn = tf.keras.losses.MeanAbsoluteError() ## a custom loss, as long as it is suitable for regression/binary classification tasks.
model.fit(train_x, train_y)

**Comment by JoshuaC3** [comment]
Looks like it would be super easy to expose this AND the optimizer in the GAMINet class.

```
        self.optimizer = tf.keras.optimizers.Adam(learning_rate=self.lr_bp)
        if self.task_type == "Regression":
            self.loss_fn = tf.keras.losses.MeanSquaredError()
        elif self.task_type == "Classification":
            self.loss_fn = tf.keras.losses.BinaryCrossentropy()
        else:
            raise ValueError("The task type is not supported")
```

I'd be happy to do a PR to add this.
